crates/ty_server/src/document/text_document.rs

```diff
@@ -16,15 +16,16 @@ pub struct TextDocument {
 
     /// The string contents of the document.
     contents: String,
-    /// A computed line index for the document. This should always reflect
-    /// the current version of `contents`. Using a function like [`Self::modify`]
-    /// will re-calculate the line index automatically when the `contents` value is updated.
-    index: LineIndex,
```

Comment:
IIUC, this is removed because the server will always need to re-compute the index whenever the content in a text document changes so there's no benefit of computing this ahead of time?

Reply:
The main reason we stored the `index` in Ruff is because we use `document.index` to map positions from text ranges to LSP ranges (and vice versa). However, this isn't the case for ty. In ty, we use `line_index` which has its own caching. 

The only remaining place where `index` was used was in the `document.update`, which we only need to compute when a document changes and the index is outdated immediately after. I didn't bother trying to use `line_index` in `document.update`. We could, but it's probably not worth the complexity.

Reply:
That makes sense, thanks for the explanation!